src/dimension_slice.c

```diff
@@ -91,12 +91,52 @@ ts_dimension_slice_cmp_coordinate(const DimensionSlice *slice, int64 coord)
 	return 0;
 }
 
+static void
+lock_result_ok_or_abort(TupleInfo *ti, DimensionSlice *slice)
+{
+	switch (ti->lockresult)
```

Comment:
If a lock was not requested (i.e., tuplock was NULL in the scan function), I assume the result will default to `TM_Ok` (since it is 0 in the enum)?

Reply:
Yes. It is initialized to all zeroes here:

```
ts_scanner_scan(ScannerCtx *ctx)
{
	InternalScannerCtx ictx = { 0 };
	TupleInfo *tinfo;

	for (ts_scanner_start_scan(ctx, &ictx); (tinfo = ts_scanner_next(ctx, &ictx));)
```

First `ictx` is zero-initialized, and then `ts_scanner_next` will return the `TupleInfo*` referencing the `TupleInfo` inside `ictx`, so it should be `TM_Ok`.